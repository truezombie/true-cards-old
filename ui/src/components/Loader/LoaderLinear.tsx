import React from 'react';
import Fade from '@material-ui/core/Fade';
import LinearProgress from '@material-ui/core/LinearProgress';
import { WithStyles, withStyles } from '@material-ui/core/styles';
import { linearLoaderStyles } from './styles';

export interface LoaderLinearProps
  extends WithStyles<typeof linearLoaderStyles> {
  show: boolean;
}

const LoaderLinear = ({ classes, show }: LoaderLinearProps): JSX.Element => {
  return (
    <Fade in={show}>
      <div className={classes.progressWrap}>
        <LinearProgress className={classes.progressWrap} />
      </div>
    </Fade>
  );
};

export default withStyles(linearLoaderStyles)(LoaderLinear);
