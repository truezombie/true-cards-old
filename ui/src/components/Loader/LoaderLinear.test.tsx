import React from 'react';
import { shallow } from 'enzyme';

import LoaderLinear, { LoaderLinearProps } from './LoaderLinear';

const classes = {
  progressWrap: 'progressWrap',
  progressHeight: 'progressHeight',
};

const setUp = (props: LoaderLinearProps) =>
  shallow(<LoaderLinear {...props} />);

describe('<LoaderLinear />', () => {
  it('should render showed LoaderLinear component', () => {
    const wrapper = setUp({
      show: true,
      classes,
    });

    expect(wrapper.html()).toMatchSnapshot();
  });

  it('should render hidden LoaderLinear component', () => {
    const wrapper = setUp({
      show: false,
      classes,
    });

    expect(wrapper.html()).toMatchSnapshot();
  });
});
