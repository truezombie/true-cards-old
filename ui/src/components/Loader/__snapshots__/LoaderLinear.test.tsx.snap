// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LoaderLinear /> should render hidden LoaderLinear component 1`] = `"<div class=\\"LoaderLinear-progressWrap-1 progressWrap\\" style=\\"opacity:0;visibility:hidden\\"><div class=\\"MuiLinearProgress-root MuiLinearProgress-colorPrimary LoaderLinear-progressWrap-1 progressWrap MuiLinearProgress-indeterminate\\" role=\\"progressbar\\"><div class=\\"MuiLinearProgress-bar MuiLinearProgress-barColorPrimary MuiLinearProgress-bar1Indeterminate\\"></div><div class=\\"MuiLinearProgress-bar MuiLinearProgress-bar2Indeterminate MuiLinearProgress-barColorPrimary\\"></div></div></div>"`;

exports[`<LoaderLinear /> should render showed LoaderLinear component 1`] = `"<div class=\\"LoaderLinear-progressWrap-1 progressWrap\\" style=\\"opacity:0\\"><div class=\\"MuiLinearProgress-root MuiLinearProgress-colorPrimary LoaderLinear-progressWrap-1 progressWrap MuiLinearProgress-indeterminate\\" role=\\"progressbar\\"><div class=\\"MuiLinearProgress-bar MuiLinearProgress-barColorPrimary MuiLinearProgress-bar1Indeterminate\\"></div><div class=\\"MuiLinearProgress-bar MuiLinearProgress-bar2Indeterminate MuiLinearProgress-barColorPrimary\\"></div></div></div>"`;
